Assets/Scripts/Round/RoundManager.cs

```diff
@@ -0,0 +1,112 @@
+        //UI element that contains the round timer
+        [SerializeField] private RectTransform timerUi = null;
+        //UI element that contains the round controls
+        [SerializeField] private RectTransform controlUi = null;
+
+        private int timerSeconds = 0;
+        private bool started = false;
+        private Coroutine tickCoroutine;
+            
+        public bool IsRoundStarted => started;
+
+        public void StartWarmup()
+        {
+            timerSeconds = warmupTimeSeconds;
+            StartCoroutine(TickWarmup());
+        }
+
+        public void StartRound()
+        {
+            started = true;
+            controlUi.gameObject.SetActive(true);
+            tickCoroutine = StartCoroutine(Tick());
+        }
+        
+        public void RestartRound()
+        {
+            if (!isServer) return;
+            
+            StopCoroutine(tickCoroutine);
+            NetworkManager.singleton.ServerChangeScene(SceneManager.GetActiveScene().name);
+        }
+        
+        private IEnumerator TickWarmup()
+        {
+            while (timerSeconds > 0)
+            {
+                RpcUpdateClientClocks(GetTimerText());
+                timerSeconds--;
+                yield return new WaitForSeconds(1);
+            }
+            
+            StartRound();
+        }
+        
+        private IEnumerator Tick()
+        {
+            //TODO: change this once we have antags/shuttle/proper round endings
+            //Restart round every 300 seconds.
+            while (timerSeconds < 300)
+            {
+                RpcUpdateClientClocks(GetTimerText());
+                timerSeconds++;
+                yield return new WaitForSeconds(1);
+            }
+            
+            RestartRound();
+        }
+
+        [ClientRpc]
+        private void RpcUpdateClientClocks(string text)
+        {
+            if(!timerUi.gameObject.activeSelf) timerUi.gameObject.SetActive(true);
+            
+            timerText.text = text;
+        }
+
+        private string GetTimerText()
+        {
+            switch (IsRoundStarted)
+            {
+                case true:
+                    return $"Round Time: {GetClockText()}";
+                case false:
+                    return $"Round Start In: {GetClockText()}";
+                default:
+                    return "Shits fucked up, yo";
+            }
+        }
+
+        private string GetClockText()
+        {
+            int hours = Mathf.FloorToInt(timerSeconds / 3600f);
+            int minutes = Mathf.FloorToInt(timerSeconds % 3600 / 60f);
+            int seconds = Mathf.FloorToInt(timerSeconds % 3600 % 60);
+
+            string hourText = hours < 10 ? $"0{hours}" : hours.ToString();
+            string minuteText = minutes < 10 ? $"0{minutes}" : minutes.ToString();
+            string secondText = seconds < 10 ? $"0{seconds}" : seconds.ToString();
+
+            return $"{hourText}:{minuteText}:{secondText}";
```

Comment:
Surely we should be using DateTime formatting?

Reply:
Indeed. Just looked it up, I should use TimeSpan class for this. Will change.